force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +103,14 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
-        }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        Integer result=0;
+        List<String> characterList = new List<String>(str.split(''));
+        for (String currentIteminLoop: characterList) {
+            if(currentIteminLoop=='a' || currentIteminLoop=='e' || currentIteminLoop=='i' || currentIteminLoop=='o' || currentIteminLoop=='u'){
+                result++; //increment the result when condition is true
```

Comment:
Yes, this works. You could also put the vowels in a set or list and use contains()